### ISSUE TITLE
Add Support for File Upload and Download

### ISSUE THREAD

**Issue by GrantErickson** [IntelliTect/Coalesce#111]
This has come up at a couple of our customer sites now. Being able to handle files naively would save significant implementation time.
Prime reason to implement is that build this is a pretty boilerplate process for uploading a file that will then be downloaded by a web page as a resource. Custom images and CSS files are examples.

These are some high level details, and I understand other parts of the feature would need to be fleshed out.

Model Example
```
class Abc {
   public int AbcId {get;set;}
   public string Name {get;set;}
   [File]
   public byte[] ImageFile {get;set;}
}
```

This would automatically create an endpoint for uploading, downloading and deleting the the content of the byte array as a file.
The [File] Attribute would likely need some parameters. 
string DefaultFilename (string of the filename to be downloaded)
string FilenameProperty (points to another property that contains the filename)
Potentially some mime type information.
Permissions: probably leverage what we already have. Might need an AllowDelete attribute property.

Add ability to upload, download, delete these from the admin pages. This is a pretty common admin activity. For example adding a set of backgrounds. 

API surface. 
1. Upload
2. Download
3. Delete

We would need to figure out exactly how the TypeScript should look. Maybe something like this.
1. Property: File URL
2. Method: Upload
3. Method: Delete

**Comment by adamskt** [comment]
If we are talking about an API controller method that will `return File(memoryStream, mimeType);` and will support 206-Partial Content range processing, I vote that we enable that feature in the default project templates as well (see (https://github.com/aspnet/Mvc/pull/6895)).  This is what you need if you are going to wire up the file content to an HTML5 <audio> or <video> tag and not force the client to download the entire file before it starts playing.

**Comment by GrantErickson** [comment]
Seems 206-partial would be reasonable to implement and would add good flexibility. We would need to work out how to use EF (or not) to read the data from the database so we only get the part we need. May want to split this into two items if this will be hard.

**Comment by ascott18** [comment]
Implemented in #116 , using a non-owned-type approach. The owned type approach should be able to be implemented in the future as a non-breaking chnage
